operators/src/upgrade/k8s/crd/v0.rs

```diff
@@ -0,0 +1,196 @@
+use crate::upgrade::k8s::crd::SEMVER_RE;
+use chrono::Utc;
+use kube::CustomResource;
+use schemars::JsonSchema;
+pub use semver::Version;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::str::FromStr;
+use validator::Validate;
+
+#[derive(
+    CustomResource,
+    Serialize,
+    Deserialize,
+    Debug,
+    Default,
+    Eq,
+    PartialEq,
+    Clone,
+    Validate,
+    JsonSchema,
+)]
+#[kube(
+    group = "openebs.io",
+    version = "v0",
+    kind = "UpgradeAction",
+    singular = "upgradeaction",
+    plural = "upgradeactions",
+    namespaced,
+    status = "UpgradeActionStatus",
+    derive = "Default",
+    derive = "PartialEq",
+    shortname = "ua",
+    printcolumn = r#"{"name":"Components state", "type":"string", "jsonPath":".status.components_state"}"#,
+    printcolumn = r#"{"name":"Target Version", "type":"string", "jsonPath":".spec.target_version"}"#,
+    printcolumn = r#"{"name":"Current Version", "type":"string", "jsonPath":".spec.current_version"}"#
+)]
+
+/// Define spec for upgrade action.
+pub struct UpgradeActionSpec {
+    /// Records the current version of the product.
+    #[validate(required, regex = "SEMVER_RE")]
```

Comment:
If all of these are "required", why not simply drop the Option?

Reply:
Ahh, we cannot remove Option since validate_required(..) expects an Option value.

Reply:
That's what I mean, if we drop the Option don't need to add required because it'll be explicitly required?

Reply:
Oh Okay. Got it.